.devcontainer/scripts/keycloak-local.js

```diff
@@ -0,0 +1,194 @@
+require('dotenv').config();
+const KcAdminClient = require('@keycloak/keycloak-admin-client').default;
+
+const KEYCLOAK_URL = process.env.KEYCLOAK_URL;
+const KEYCLOAK_USERNAME = process.env.KEYCLOAK_USER;
+const KEYCLOAK_PASSWORD = process.env.KEYCLOAK_PASSWORD;
+
+const KEYCLOAK_REALM = process.env.KEYCLOAK_REALM;
+const KEYCLOAK_CLIENT_ID = process.env.KEYCLOAK_CLIENT_ID;
+const KEYCLOAK_CLIENT_SECRET = process.env.KEYCLOAK_CLIENT_SECRET;
+
+const ID_PROVIDER = process.env.ID_PROVIDER;
+const ID_PROVIDER_CLIENT_ID = process.env.ID_PROVIDER_CLIENT_ID;
+const ID_PROVIDER_CLIENT_SECRET = process.env.ID_PROVIDER_CLIENT_SECRET;
+
+const kcAdminClient = new KcAdminClient({ baseUrl: `${KEYCLOAK_URL}/auth` });
+
+async function main() {
+  try {
+    await kcAdminClient.auth({
+      grantType: 'password',
+      clientId: 'admin-cli',
+      password: KEYCLOAK_PASSWORD,
+      username: KEYCLOAK_USERNAME,
+    });
+
+    const realms = await kcAdminClient.realms.find();
+
+    const existingRealm = realms.find(realms => realms['realm'] === KEYCLOAK_REALM);
+
+    if (!existingRealm) {
+      const createRealm = (realm) => kcAdminClient.realms.create({ realm, enabled: true });
+      const payload = await createRealm(KEYCLOAK_REALM);
+      console.log("Realm created: ", payload);
+    }
+
+    kcAdminClient.setConfig({
+      realmName: KEYCLOAK_REALM,
+    });
+
+    const clients = await kcAdminClient.clients.find({ realm: KEYCLOAK_REALM });
+    const client = clients.find((x) => x.clientId === KEYCLOAK_CLIENT_ID);
+
+    if (!client) {
+      console.log(`Creating client: ${KEYCLOAK_CLIENT_ID}`);
```

Comment:
Are all console.logs intentional?

Reply:
yeah it was, just gives some feedback when running that script from the terminal